test/json-filter.test.js

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+
+const assert = require('assert');
+const { Response } = require('@adobe/helix-universal');
+const jsonFilter = require('../src/json-filter.js');
+
+const TEST_DATA = [];
+for (let i = 0; i < 50; i += 1) {
+  const row = {};
+  TEST_DATA.push(row);
+  for (let j = 0; j < 4; j += 1) {
+    row[`col${j}`] = `cell(${i},${j})`;
+  }
+}
+
+const TEST_SINGLE_SHEET = {
+  offset: 0,
+  limit: TEST_DATA.length,
+  total: TEST_DATA.length,
+  data: TEST_DATA,
+};
+
+const TEST_MULTI_SHEET = {
+  ':names': ['sheet1', 'sheet2'],
+  sheet1: TEST_SINGLE_SHEET,
+  sheet2: TEST_SINGLE_SHEET,
+};
+
+describe('JSON Filter test', () => {
+  it('returns same response for single sheet with no query', async () => {
+    const dataResponse = new Response(JSON.stringify(TEST_SINGLE_SHEET), {
+      headers: {
+        'content-type': 'application/json',
+      },
+    });
+
+    const filter = jsonFilter({}, console);
+    const resp = await filter(dataResponse);
+    assert.strictEqual(resp.status, 200);
+    assert.deepStrictEqual(await resp.json(), {
+      offset: 0,
+      limit: TEST_DATA.length,
+      total: TEST_DATA.length,
+      data: TEST_DATA,
+    });
+    assert.deepStrictEqual(resp.headers.plain(), {
+      'content-type': 'application/json',
```

Comment:
I don't think it is ideal to have the metadata in the JSON in one layer and in the headers in another. This makes the format more confusing and harder to validate.

Reply:
after talking to @davidnuescheler, we agreed to move the metadata to headers.
I added it in both places to be backward and forward compatible.

but we can remove the old schema and start using the new one. i.e. no metadata in the json.